src/books/show/route.js

```diff
@@ -6,16 +6,18 @@ import storage from '../storage';
 export default Route.extend({
   initialize(options = {}) {
     this.layout = options.layout;
+    this.listenTo(this, 'fetch', this.onFetch);
   },
 
-  fetch() {
+  fetch(id) {
+    this.id = id;
     return storage.findAll().then(collection => {
       this.collection = collection;
     });
   },
 
-  onFetch(id) {
-    this.model = this.collection.get(id);
+  onFetch() {
+    this.model = this.collection.get(this.id);
     this.collection.active = this.model;
```

Comment:
just move this to the `fetch()` function above inside the `then(collection => { ... });`